### ISSUE TITLE
New plot flag: allow visiting of untrusted players

### ISSUE THREAD

**Issue by mindw0rm** [IntellectualSites/PlotSquaredSuggestions#64]
It would be nice if you could allow players to visit a plot without having to add/trust them.

There is already the permission `plots.visit.other`, but that will grant visit rights to _every_ plot, no matter if the owner wants it or not.

A new plot flag would enable this on a per-plot basis, allowing the owner to control visit rights. 

Note that I already have a [solution](https://github.com/mindw0rm/PlotSquared/tree/UntrustedVisit), that is running on our server since some weeks. 
You'll get a pull request soon.

**Comment by RedstoneFuture** [comment]
Already implemented as "deny-teleport":
https://github.com/IntellectualSites/PlotSquared/wiki/Flag:-deny-teleport

**Comment by mindw0rm** [comment]
No, deny-teleport is completely different.

TL;DR:  deny-teleport _denies_ certain (i.e. most) teleports, while untrusted-visit _allows_ some teleports that are otherwise forbidden (even without deny-teleport). 

In detail: 
If _neither_ flag is set, 
* owners, added and trusted players (added only if the owner is online) can use `/p visit` to teleport to the plot('s home location), _untrusted players can not visit_. (this is simplified, since owners and added/trusted players still need a certain permission, and there is also a permission to allow others to visit, but that would be global and not plot-specific.)
* everyone can use other means of teleportation, vanilla examples would be chorus fruits or ender pearls. Other plugins also allow some teleportation (like `/home` or `/back` from CMI).

If deny-teleport is set, _all_ teleportation on or within the plot is forbidden (or should be, but this is currently buggy, see IntellectualSites/PlotSquared#2504 and IntellectualSites/PlotSquared#2505).

If untrusted-visit is set (as I implemented it in IntellectualSites/PlotSquared#2542), _everyone_ can use `/p visit` to visit the plot. _Even_ if deny-teleport is set.

**Comment by MattBDev** [comment]
I don't see why such a flag is needed. In what case do you only want certain players to teleport to a plot but not have them trusted or added?

**Comment by mindw0rm** [comment]
I'm doing a labyrinth, with a price at the end. I do **not** want to trust anyone, because they would simply cheat by breaking down walls etc. 
I also do *not* want to give the permission plots.visit.other to everyone, because there are others on the server (whose players are mainly kids <= 12 yo) who explicitly do _not_ want that everyone can easily visit their plot. 
But I want to provide them with a simply way to reach my plot, without having to run over half the map. 

I _could_ set up a portal or warp point or such, but that would be abuse of admin powers; it is preferable if everyone has that ability by simply setting a flag. Giving the permission to set the flag is safe since they can only affect their plot, and they don't have to use it. 
(also it would be nice if we could one day return to the original PlotSquared ;) )

The whole thing is about increase of flexibility regarding the configuration of plots. 

Of course this is already possible on our server since we use my fork, but I thought since it is such a simple change without much potential to cause side effects (I am unable to see how this could effect anything what it shouldn't), it would be nice to contribute so that others can benefit from it as well. 

[btw. this suggestion here only exists so that I could put down an issue in my pull request]

**Comment by AeSix** [comment]
Honestly, this seems out of scope for what PlotSquared was intended to be and do. 

With that said, this sounds like it would be a great idea for an add-on.

**Comment by mindw0rm** [comment]
Honestly, I can't see how to realize this as an add-on, since it has to change something deep into the code (did you even look at the implementation?).  How can I even un-prevent a teleport/command use if you already prevented it? 

I also disagree that this is out of scop, since it olny improves _existing_ functions. 


But okay, I understand that you have no interest in refined functionality, and I won't bother you with any more imrovements I might make, and keep using the fork :(

**Comment by MattBDev** [comment]
@mindw0rm You can submit the PR and I can review it. I just don't know if it is something I will accept it right away. I am going to reopen this for further discussion and I really hope to see the PR soon.

**Comment by AeSix** [comment]
@mindw0rm I toss my opinion in when I have one.  Not to discourage anyone, but to bring a different view point.  I'm sorry if I upset you.

My only real consideration is the amount of complexity which has creeped into PS over the years, with user confusion increasing exponentially.  That, and the amount of bugs currently still in PlotSquared which can be compounded by additional features.  

Re-reading what you're wanting, you may be right in that it would fail as an add-on.  However, it does also sound like a wonderful candidate for plots marked as server plots.   Again, just my opinions.

**Comment by mindw0rm** [comment]
@MattBDev The PR was already there but I closed it after three of you made clear that this is something you do not want. 
Well, now it is reopened.
